### PR TITLE
Convert loop and lightweight example to component yaml

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Part of #670

**Description of your changes:**
Nested loop and lightweight component are examples that doesn't depend on any external component and sdk changes. Thus moving them to follow the component.yaml format.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by Tomcli** [comment]
/hold

wait for 0.9.0 release

File: samples/lightweight-component/calc_pipeline.py [hunk 1]
```diff
@@ -57,18 +57,18 @@ def divmod_helper(dividend, divisor):
...
-   name='Calculation pipeline',
+   name='calculation-pipeline',
    description='A toy pipeline that performs arithmetic calculations.'
 )
 # Currently kfp-tekton doesn't support pass parameter to the pipelinerun yet, so we hard code the number here
 def calc_pipeline(
-   a='7',
-   b='8',
-   c='17',
+   a:float = '7',
```
Comment (yhwang) [inline samples/lightweight-component/calc_pipeline.py:65]: nit: because its data type is float, I guess removing the quote would be more intuitive.

Comment (Tomcli) [reply samples/lightweight-component/calc_pipeline.py]: thanks, updated

**Review by yhwang — APPROVED** [review]
/lgtm

**Pull request merged**